source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -686,6 +691,11 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       case CheckResult::OK:
         ENVOY_STREAM_LOG(trace, "'{}':'{}'", *decoder_callbacks_, key, value);
         request_headers_->setCopy(Http::LowerCaseString(key), value);
+        if (!headersWithinLimits(*request_headers_)) {
```

Comment:
Is it possible that we move check before adding those headers, and reject them earlier. it is probably not a big deal but cloud be an optimization if it is easy to do so

Reply:
This is how ext_proc does it and it's a lot simpler. That's why I chose to do it this way.

Pre-check for mutations depends on reproducing implementation details of the header map (e.g. add header vs set, existing header vs new, multi-value headers, inline headers, etc...) so it's a lot more complicated and imo brittle. I think we should just apply the mutation and do a post-check.

Reply:
Sure SG! as i mentioned above, it is a small optimization if it is easy to do :)